matchmake-extension/get_simple_playing_session.go

```diff
@@ -28,8 +28,9 @@ func (commonProtocol *CommonProtocol) getSimplePlayingSession(err error, packet
 	}
 
 	simplePlayingSessions := make(map[string]*match_making_types.SimplePlayingSession)
-
-	for gatheringID, session := range common_globals.Sessions {
+	
+	var errRet *nex.Error = nil
+	common_globals.EachSession((func(gatheringID uint32, session *common_globals.CommonMatchmakeSession) bool {
```

Comment:
`EachSession` returns a bool if the loop exited early. In this case the only time it would exist early is if an error occurred. This can be used to remove the need to define the extra `errRet` variable

Such as:

```go
if common_globals.EachSession(func(gatheringID uint32, session *common_globals.CommonMatchmakeSession) bool {
	for _, pid := range listPID.Slice() {
		key := fmt.Sprintf("%d-%d", gatheringID, pid.Value())

		if simplePlayingSessions[key] == nil {
			connectedPIDs := make([]uint64, 0)

			session.ConnectionIDs.Each(func(_ int, connectionID uint32) bool {
				player := endpoint.FindConnectionByID(connectionID)
				if player == nil {
					common_globals.Logger.Warning("Player not found")
					return false
				}
				connectedPIDs = append(connectedPIDs, player.PID().Value())
				return false
			})

			if slices.Contains(connectedPIDs, pid.Value()) {
				attribute0, err := session.GameMatchmakeSession.Attributes.Get(0)
				if err != nil {
					common_globals.Logger.Error(err.Error())
					return true
				}

				simplePlayingSessions[key] = match_making_types.NewSimplePlayingSession()
				simplePlayingSessions[key].PrincipalID = pid.Copy().(*types.PID)
				simplePlayingSessions[key].GatheringID = types.NewPrimitiveU32(gatheringID)
				simplePlayingSessions[key].GameMode = session.GameMatchmakeSession.GameMode.Copy().(*types.PrimitiveU32)
				simplePlayingSessions[key].Attribute0 = attribute0.Copy().(*types.PrimitiveU32)
			}
		}
	}

	return false
}) {
	nex.NewError(nex.ResultCodes.Core.InvalidArgument, "change_error")
}
```

This is already the pattern being introduced in the [friends server rewrite](https://github.com/PretendoNetwork/friends/blob/e2a630d1a761356ae3f360258fa021e519dca92c/nex/friends-3ds/sync_friend.go#L56-L76)

Reply:
Except that if a new error is needed to be introduced, the pattern will have to be changed again.

Reply:
Correct. But that's not currently the case, and I don't see it needing to be the case any time soon, if ever? There's only ever one area where this snippet will error out and need to bail early, therefore only one kind of error needs to be handled. If something changes and more areas have the potential to throw errors then that will likely come from other refactors, which then would warrant a pattern change. But currently this can fit within the existing patterns